src/resizer/include/resizer/Resizer.hh

```diff
@@ -121,6 +125,8 @@ public:
 			    LibertyCell *buffer_cell);
   // Area of the design in meter^2.
   double designArea();
+  // Invalidate incremental designArea
+  void designAreaInvalid();
```

Comment:
using an invalid flag is not consistent with the way the resizer uses design_area.
you can't check the area after adding an instance if it means adding them up
for every instance. that is n^2. that is why the resizer incrementally maintains 
design_area_

Reply:
We just need a way for other tools to update the design area, other possible options:
- Exposing setDesignArea command but the disadvantage is that there might be multiple sources of truth for area calculation. 
- Exposing a resetArea command that would let the resizer re-calculate the design_area but it will be extra runtime whenever it is called, although I don't think it would be called frequently.

Or what do you suggest?

Ideally, I believe it should be part of OpenDB itself..

Reply:
Ahmed, I think this question is at the root of how we can leverage the resizer vs creating an alternate infrastructure. Take a close look at how the design area is kept up to date incrementally. We can discuss at our sync up.

Reply:
I see that design_area_ is protect so you could derive from the class and add a new transform and access the variable to keep it up to date like the other transforms do. I am not sure this is the best way to go but its an example of how to add minimal code and add the new transforms.

Reply:
I use incremental a similar incremental design_area variable too in OpenPhySyn, the problem is that if OpenPhySyn (or any other tool, e.g. CTS) make any update during the **same** session, the **report_design_area** will not be valid which is why we need a method to invalidate/update area. As suggested above we can expose a **setDesignArea** method ( or make Psn a friend class, or inheritance as you suggested), the only drawback is as I mentioned there will be multiple tools that are able to set design area which is not an optimal practice, which is why I believe it should be handled incrementally (and only) from OpenDB itself, but I am fine with any of the suggested solutions for now, we can discuss further in our meeting.

Reply:
Ideally we would make this a db operation and edit via the db and have listeners. We are far from that architecture now. If you concentrate on the transforms that you have now I think all of them can work under the inheritance model. You are right that there will be N pieces of code that add or subtract from the design area versus n pieces of code editing the netlist but it should be workable for a long time. I am the evangelist for the shared DB model but in this case need to be a pragmatist. 

In terms of the same session, its not serial operations in the same session but serial operations in the same optimization step. The number of combinations there is much lower.  Once the code works with unit tests to make sure the calculation is correct it will be very stable.

Reply:
I believe inheritance won't work (compile) since I cannot access a protected member of another instance of the class (even if the calling classes inherits from it, it can only call methods on the base class of itself, not another instance); let's discuss in the sync easier.